sync/metrics.go

```diff
@@ -22,10 +23,28 @@ func newMilliTimer(sum prometheus.Summary) *prometheus.Timer {
 	}))
 }
 
+func newFetchRequestTimer(msgtype server.MessageType) *prometheus.Timer {
```

Comment:
I don't like the metrics package depending on stuff, as any other package might be dependent on metrics.. this can easily turn into a dependency cycle.

Reply:
so I would just take this method out of this file

Reply:
this is in the sync package why the problem ?